src/components/App.tsx

```diff
@@ -1,49 +1,53 @@
-import React, { useEffect } from "react";
-import { DragDropContext, Droppable } from "react-beautiful-dnd";
-import { useSelector } from "react-redux";
+import { useEffect } from "react";
+import { DragDropContext, DragUpdate, Droppable } from "react-beautiful-dnd";
+import { useAppDispatch, useAppSelector } from "../app/hooks";
 import List from "./List/main/List";
 import AddList from "./List/AddList/AddList";
 import NavBar from "./NavBar/NavBar";
 import BoardOptionsMenu from "./BoardOptionsMenu/BoardOptionsMenu";
-import { useDispatch } from "react-redux";
 import { updateAllLists } from "../features/listsSlice";
 import { updateAllCards } from "../features/cardsSlice";
 import CardModal from "./CardModal/CardModal";
 import ListOptionsMenu from "./List/ListOptionsMenu/ListOptionsMenu";
 import { setNewBoardState } from "../features/boardSlice";
 
-const App = () => {
-  const dispatch = useDispatch();
-  const board = useSelector((state) => state.board.value);
-  const lists = useSelector((state) => state.lists.value);
-  const cards = useSelector((state) => state.cards.value);
-  const showBoard = useSelector((state) => state.boardOptions.value);
-  const currentSelectedCard = useSelector((state) => state.currentSelectedCard.value);
-  const currentSelectedList = useSelector((state) => state.currentSelectedList.value);
+const App: React.FunctionComponent = () => {
+  const dispatch = useAppDispatch();
+  const board = useAppSelector((state) => state.board.value);
+  const lists = useAppSelector((state) => state.lists.value);
+  const cards = useAppSelector((state) => state.cards.value);
+  const showBoard = useAppSelector((state) => state.boardOptions.value);
+  const currentSelectedCard = useAppSelector((state) => state.currentSelectedCard.value);
+  const currentSelectedList = useAppSelector((state) => state.currentSelectedList.value);
 
   useEffect(() => {
-    if (!board.boardId) {
+    if (!board.id) {
       dispatch(setNewBoardState());
     }
 
-    document.getElementById("lists-wrapper").scrollLeft = 0;
+    const listWrapper = document.getElementById("lists-wrapper");
+    if (listWrapper) listWrapper.scrollLeft = 0;
   }, []);
 
-  const handleDragEnd = (result) => {
+  const handleDragEnd = (result: DragUpdate) => {
     const { destination, source, draggableId, type } = result;
     const newListsOrder = [...lists];
 
     if (!destination) return;
 
     if (type === "list") {
       const listToMove = newListsOrder.find((list) => list.id === draggableId);
-
-      newListsOrder.splice(source.index, 1);
-      newListsOrder.splice(destination.index, 0, listToMove);
+      if (listToMove) {
+        newListsOrder.splice(source.index, 1);
+        newListsOrder.splice(destination.index, 0, listToMove);
+      }
 
       dispatch(updateAllLists(newListsOrder));
     } else {
       const draggingCard = cards.find((card) => card.id === draggableId);
+      if (!draggingCard) {
+        return;
+      }
```

Comment:
In a lot of places, I've had to add these explicit checks to see if something is defined as typescript complains its possibly null. This is fine for now but I'll be raising a ticket to investigate how better to handle this